File: solutions/c/eliuds-eggs/1/eliuds_eggs.c
#include "eliuds_eggs.h"

int egg_count(unsigned int display_value) {
  int result = 0;
  int bit = 0;

  while (display_value) {
    bit = display_value & 1;
    result += bit;
    display_value = display_value >> 1;

  }

  return result;
}

File: solutions/typescript/eliuds-eggs/1/eliuds-eggs.ts
//
// This is only a SKELETON file for the 'Pop Count' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const eggCount = (displayValue: number): number => {
  let eggs = 0
  displayValue.toString(2).split('').forEach((bit) => {
    if (bit === '1') {
      eggs++
    }
  })
  return eggs
}
